plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableParameter.java

```diff
@@ -184,10 +190,11 @@ private Object uncoerce(Object o, Type type) {
         } else if (o != null && !o.getClass().getName().startsWith("java.")) {
             try {
                 // Check to see if this can be treated as a data-bound struct.
-                Map<String, Object> nested = DescribableModel.uninstantiate_(o);
+                UninstantiatedDescribable nested = DescribableModel.uninstantiate2_(o);
                 if (type != o.getClass()) {
-                    nested.put(CLAZZ, o.getClass().getSimpleName());
+                    nested.setKlass(o.getClass().getSimpleName());
                 }
+                nested.setSymbol(symbolOf(o));
```

Comment:
@kohsuke feels it is clearer to require the symbol on each element. When picking that symbol, we may prefer to use a concise name, since it is not used to disambiguate, and the full meaning is clear from the parameter name; for example:

```groovy
checkout(git(userRemoteConfigs: [
  config(url: '…', credentialsId: '…')]))
```